### PR TITLE
Save user location via dashboard

### LINKED ISSUE #8566: (multiple issues) Fix UI flow and debug dashboard location input form
We have a feature where you can add your location from the dashboard -- https://publiclab.org/dashboard (while logged in):

image

You click "Add your location" and a modal popup appears, prompting you:

image

Unfortunately, as @Tlazypanda noted, this doesn't really work smoothly, both due to possible UI design gaps (no notification to the user when tags are created) and possibly bugs. @Tlazypanda noted:

> * clicking "Save" doesn't apparently do anything, because you're just dropped back on the dashboard
> * On adding location modal, the slider works by zooming in but when we save it and check it out in the profile page it is saved without any zoom.
> * Also after adding your location it still shows add location on dashboard
> * This is invoked by adding location from dashboard but on clicking share location from profile page nothing happens.
> * Also after adding location for first time, it doesn't seem to update for a second time

Even after fixing this, we also ought to add a system test for adding profile tags - like, one of these modified for the dashboard!

https://github.com/publiclab/plots2/blob/86776c2c72a505b0db563f52d1a8165c36c6eb37/test/system/post_test.rb#L188-L211

We could add one of these assertions afterwards that checks that the profile tags are properly created: 

https://github.com/publiclab/plots2/blob/86776c2c72a505b0db563f52d1a8165c36c6eb37/test/system/tag_test.rb#L40-L51

### Intended workflow

The ideal intended workflow is:

1. See "add a location" on dashboard, click it
2. See modal and check content of one or more of the inputs
3. enter lat/lon and zoom, or use the slider
4. press `Save` and see an alert (maybe?) that says "profile location was saved. Learn more about location privacy" - linking to https://publiclab.org/location-privacy
5. confirm dashboard button now says "View nearby content
6. confirm tags were added and appear on profile page

### PR DESCRIPTION
**Pull request by gauravahlawat81**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below
Screencast-from-11-01-21-041344

### CODE REVIEW

File: app/views/tag/_location.html.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@
       <% end %>
       if( $(this).attr("submit_to") ) {
         modalUrl += firstVar ? '?' : '&';
-        modalUrl += 'url=' + encodeURIComponent(locationInput.attr("submit_to"));
+        modalUrl += 'url=' + encodeURIComponent($(this).attr("submit_to"));
```
Comment (coder645) [inline app/views/tag/_location.html.erb:18]: @gauravahlawat81 I didn't quite understand what you are doing here. Why did you use `$(this)` ?

**Comment by coder645** [comment]
Also @gauravahlawat81 I'm not very sure if this is a valid PR because I'm not able to reproduce this bug locally. Saving the location through the dashboard displays it on the profile for me! So I'm not really sure what you are tryin to do through this PR.

@jywarren @cesswairimu any thoughts?

**Comment by gauravahlawat81** [comment]
@coder645  please see this video, [Video Link](https://drive.google.com/file/d/1sGC66b978YjP0CCyBB8GW_2A6IkgUPCR/view?usp=sharing). Adding the location through dashboard doesn't save the location for me. I checked through rails console also and no location tag were added there as well.

File: app/views/tag/_location.html.erb (hunk 1 above)
Comment (gauravahlawat81) [reply app/views/tag/_location.html.erb:18]: I don't know why, but this doesn't seem to work, when we select using `locationInput` it returns undefined. So I though if we are searching if `submit_to` exists or not using `$(this)`, we can also use the value in same way,

Please let me know if you have any better ideas.

Comment (coder645) [reply app/views/tag/_location.html.erb:18]: I don't have any better ideas per se...but I wasn't too sure how this was working.
Anyway thanks for explaining.
Cheers!

**Comment by jywarren** [comment]
Hmm, that's interesting -- @coder645 can you make a recording of the steps your taking too?
 
@gauravahlawat81 and @coder645 i wonder if you've found an edge case or something... i'd love to better understand what is causing the discrepancy between what you two are each seeing before merging this! Maybe we can write or adjust a system test to cover all cases, as well?

Not to mention your notes above on the code being hard to read suggests that we should try to leave comments or otherwise try to improve the code legibility! It's hard to know what to fix or if it's properly fixed without clearly being able to understand the code and it's purpose and methods!

Many thanks to both of you!

**Comment by Manasa2850** [comment]
@gauravahlawat81 even I'm not able to save the location through the dashboard. I tried making the change you've made i.e changing `locationInput` to `$(this)` but it still doesn't save the location for me :(
ezgif com-video-to-gif

CC: @jywarren

**Comment by gauravahlawat81** [comment]
@Manasa2850 ,can you check through your rails console ? I guess location is being saved for you, otherwise you would see `Share your Location` on the profile page. If you run `UserTag.all`  in rails console, it should show you the location you saved, and if not then we'll have to investigate further

**Comment by Manasa2850** [comment]
Oh yes you're right @gauravahlawat81 it's getting reflected in the rails console! Not sure why it's not seen in the profile page. I'll look into it. Thanks :+1:

**Comment by coder645** [comment]
@gauravahlawat81 @Manasa2850 I don't see the `Share your Location` button even without making the change @gauravahlawat81 has made. So I'm assuming it saves the location even without the change made.
Cheers!

**Comment by gauravahlawat81** [comment]
@coder645 Can you run `UserTag.delete_all` in your rails console.This will ensure that all previous tags are deleted and you can start fom scratch.After that can you please post the GIF of the problem you are facing ? You also said that you could save the location without making any changes, if you could, please upload the GIF of these steps too. Thanks!

**Comment by RuthNjeri** [comment]
Fixed via https://github.com/publiclab/plots2/pull/8983

**Comment by RuthNjeri** [comment]
Thanks, @gauravahlawat81 🎉
